client/test/account_store.spec.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+const assert = require('chai').assert;
+
+const mockeryUtils = require('./utils/mockery_utils');
+const SpecDispatcher = require('./utils/specs_dispatcher');
+const ActionTypes = require('../app/constants/app_constants').ActionTypes;
+
+
+describe('Account Store', () => {
+  let accountStore;
+  let dispatcher;
+  let account1;
+  let account2;
+  let account3;
+
+  function addAcount(account) {
+    dispatcher.dispatch({
+      type: ActionTypes.ADD_ACCOUNT_SUCCESS,
+      value: { account },
+    });
+  }
+
+  before(() => {
+    dispatcher = new SpecDispatcher();
+    mockeryUtils.initDispatcher(dispatcher);
+    mockeryUtils.initForStores();
+    accountStore = require('../app/stores/account_store');
+  });
+
+  before(() => {
+    account1 = accountStore.makeEmptyAccount().toObject();
+    account2 = accountStore.makeEmptyAccount().toObject();
+    account3 = accountStore.makeEmptyAccount().toObject();
+    account1.id = '123';
+    account1.label = 'personal';
+    account1.mailboxes = [
+      { id: 'a1', label: 'inbox', attribs: '', tree: [''], accountID: '123' },
+      { id: 'a2', label: 'sent', attribs: '', tree: [''], accountID: '123' },
+    ];
+    account2.id = '124';
+    account2.label = 'pro';
+    account2.mailboxes = [
+      {
+        id: 'b1', label: 'mailbox', attribs: '', tree: [''], accountID: '124',
+      },
+      {
+        id: 'b2', label: 'folder1', attribs: '', tree: [''], accountID: '124',
+      },
+    ];
+    account3.id = '125';
+    addAcount(account1);
+    addAcount(account2);
+  });
+
+
+  /*
+   * Problem noticed:
+   *
+   * * Remove mailbox event doesn't do anything.
+   * * getDefault can be used as getByMailbox.
+   */
+  describe('Actions', () => {
+    it('ADD_ACCOUNT_SUCCESS', () => {
+      dispatcher.dispatch({
+        type: ActionTypes.ADD_ACCOUNT_SUCCESS,
+        value: { account: { id: '125' } },
+      });
+      const accounts = accountStore.getAll();
+      assert.equal(accounts.get('125').get('id'), '125');
+      assert.deepEqual(accounts.get('125').get('mailboxes').toObject(), {});
```

Comment:
To complement the previous point, using fixtures will permit a more clever way to deep compare contents,